### ISSUE TITLE
Include LINE pragmas in generated output?

### ISSUE THREAD

**Issue by edsko** [well-typed/hs-bindgen#74]
Not sure quite how useful this is, but we could consider adding `LINE` pragmas to the generated output to relate the generated headers back to the C headers. However, it would mean that the user will get `ghc` errors or warnings about (seemingly) C code, which might actually be more confusing than helpful. But worth experimenting with.

Related: #23.

**Comment by phadej** [comment]
Probably not useful. We generate code quite out-or-order, and we don't include user written code, so GHC cannot really complain about anything user has written. (`hs-bindgen` can complain on something wrong with header file, but that's *before* we produce any output).

**Comment by edsko** [comment]
Yes, I agree that for error messages it would not be helpful. But perhaps it'll be useful for someone looking through the generated code and trying to understand how it relates back to the C header? But maybe that's better achieved through haddocks (#26) rather than LINE pragmas?

**Comment by phadej** [comment]
> But perhaps it'll be useful for someone looking through the generated code and trying to understand how it relates back to the C header?

We have https://github.com/well-typed/hs-bindgen/issues/316

**Comment by edsko** [comment]
Closing this; as Oleg points out, errors in the generated code should not refer back to the C source (instead we should catch the error in the C source ourselves; any remaining errors in the generated Haskell code would then be bugs in `hs-bindgen`). Instead we should add the links to the C code in the Haddocks; added this to #26.
